infra/core/Makefile

```diff
@@ -0,0 +1,211 @@
+	@command -v docker-compose >/dev/null 2>&1 || { echo "❌ Docker Compose is required but not installed."; exit 1; }
+	@command -v yarn >/dev/null 2>&1 || { echo "❌ Yarn is required but not installed."; exit 1; }
+	@echo "✅ All prerequisites installed."
+	@echo ""
+	@echo "✅ Setup complete! Run 'make start' to launch the core stack."
+
+build:
+	@echo "🔨 Building core indexers..."
+	@echo "  Using RPC: $${RPC_URLS}"
+	@echo "  Using Chain ID: $${CHAIN_ID}"
+	@echo "Checking if project.yaml files need regeneration..."
+	@cd ../../indexers; \
+	if [ ! -f consensus/project.yaml ] || \
+	   [ ! -f files/project.yaml ] || \
+	   ! grep -q "$${RPC_URLS}" consensus/project.yaml 2>/dev/null || \
+	   ! grep -q "$${RPC_URLS}" files/project.yaml 2>/dev/null || \
+	   ! grep -q "$${CHAIN_ID}" consensus/project.yaml 2>/dev/null || \
+	   ! grep -q "$${CHAIN_ID}" files/project.yaml 2>/dev/null; then \
+		echo "Environment variables changed or project.yaml missing - forcing regeneration..."; \
+		rm -f consensus/project.yaml files/project.yaml; \
+	fi
+	@cd ../../indexers && \
+		RPC_URLS="$${RPC_URLS}" CHAIN_ID="$${CHAIN_ID}" START_BLOCK_CONSENSUS="$${START_BLOCK_CONSENSUS}" START_BLOCK_FILES="$${START_BLOCK_FILES}" \
+		yarn lerna run codegen --scope=consensus --scope=files
+	@cd ../../indexers && \
+		RPC_URLS="$${RPC_URLS}" CHAIN_ID="$${CHAIN_ID}" START_BLOCK_CONSENSUS="$${START_BLOCK_CONSENSUS}" START_BLOCK_FILES="$${START_BLOCK_FILES}" \
+		yarn lerna run build --scope=consensus --scope=files || \
+		(echo "⚠️  Lerna/Nx cache error occurred, but checking if build succeeded..." && \
+		 if [ -f consensus/dist/index.js ] && [ -f files/dist/index.js ]; then \
+			echo "✅ Build artifacts exist - continuing..."; \
+		 else \
+			echo "❌ Build failed - no artifacts found"; exit 1; \
+		 fi)
+	@echo "✅ Build complete!"
+
+force-build:
+	@echo "🔨 Force building core indexers (clearing cache and rebuilding)..."
+	@echo "  Using RPC: $${RPC_URLS}"
+	@echo "  Using Chain ID: $${CHAIN_ID}"
+	@echo "Clearing all caches and generated files..."
+	@cd ../../indexers && rm -rf .nx/cache node_modules/.cache consensus/project.yaml files/project.yaml consensus/dist files/dist
+	@echo "Building consensus indexer..."
+	@cd ../../indexers/consensus && \
+		RPC_URLS="$${RPC_URLS}" CHAIN_ID="$${CHAIN_ID}" START_BLOCK_CONSENSUS="$${START_BLOCK_CONSENSUS}" \
+		yarn codegen && yarn build
+	@echo "Building files indexer..."
+	@cd ../../indexers/files && \
+		RPC_URLS="$${RPC_URLS}" CHAIN_ID="$${CHAIN_ID}" START_BLOCK_FILES="$${START_BLOCK_FILES}" \
+		yarn codegen && yarn build
+	@echo "✅ Force build complete!"
+
+start: build
+	@echo "🚀 Starting Core Stack..."
+	@docker-compose -p $(COMPOSE_PROJECT) -f $(COMPOSE_FILE) up -d
+	@echo ""
+	@echo "🔧 Initializing services..."
+	@$(MAKE) init-hasura
+	@echo ""
+	@echo "✅ Core stack started!"
+	@echo ""
+	@echo "📊 Services:"
+	@echo "  - Hasura GraphQL: http://localhost:8080"
```

Comment:
Wouldn't this endpoints depend on environmental variables?

Reply:
fixed